shaders/src/shading_parameters.fs

```diff
@@ -33,7 +33,13 @@ void computeShadingParams() {
 #endif
 
     shading_position = vertex_worldPosition;
-    shading_view = normalize(frameUniforms.cameraPosition - shading_position);
+    if (frameUniforms.clipFromViewMatrix[2].w == -1.0) {
```

Comment:
is it OK to test for floating point equality?

Reply:
`frameUniforms.clipFromViewMatrix[2].w` will always come from a floating point literal in the C++ code, which [uses IEEE-754 floats](https://en.cppreference.com/w/cpp/language/types#Floating-point_types). Metal uses IEEE-754 too (see [Table 2.1](https://developer.apple.com/metal/Metal-Shading-Language-Specification.pdf)) and [so does Direct3D 11](https://docs.microsoft.com/en-us/windows/win32/direct3d11/floating-point-rules). So unless `matc` messes up sth it is safe to compare against these literals